### PR TITLE
Skeleton + CertificateReissuance sub stucture + TestMenu(EXPOSUREAPP-11886, EXPOSUREAPP-11941, EXPOSUREAPP-11908)

### PR DESCRIPTION
**Pull request by LukasLechnerDev**
### This PR contains
- Extension of the DccWalletInfo Model with CertificateReissuance (I also copied the sample json from the techspec for serialization and deserialization tests.)
- Test Menu
- Skeleton (some tasks are still open, see below)

### Open Skeleton Tasks: 
- [x] Create fragment placeholders
- [x] Prepare NavGraph
- [x] Add dummy data for tile in `PersonCertificateProvider`
- [x] Repositories: add replace function with empty body

### CODE REVIEW

File: Corona-Warn-App/src/main/res/navigation/covid_certificates_graph.xml [hunk 1]
```diff
@@ -296,4 +299,9 @@
         android:name="de.rki.coronawarnapp.covidcertificate.person.ui.admission.AdmissionScenariosFragment"
         android:label="fragment_admission_scenarios"
         tools:layout="@layout/fragment_admission_scenarios" />
+    <fragment
+        android:id="@+id/DccReissuanceConsentFragment"
```
[the remark this answers is not in the document]
Comment (mtwalli) [reply Corona-Warn-App/src/main/res/navigation/covid_certificates_graph.xml:303]: can you add the success screen too ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/core/TestCertificateRepository.kt [hunk 2]
```diff
@@ -533,6 +533,13 @@ class TestCertificateRepository @Inject constructor(
         }
     }
 
+    suspend fun replaceCertificate(
+        certificateToReplace: TestCertificateContainerId,
+        newCertificate: TestCertificateContainerId
```
[the remark this answers is not in the document]
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/core/TestCertificateRepository.kt:538]: Changed.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/VaccinationRepository.kt [hunk 3]
```diff
@@ -434,6 +434,13 @@ class VaccinationRepository @Inject constructor(
         }
     }
 
+    suspend fun replaceCertificate(
+        certificateToReplace: VaccinationCertificateContainerId,
+        newCertificate: VaccinationCertificateContainerId
```
[the remark this answers is not in the document]
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/VaccinationRepository.kt]: Changed.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/success/DccReissuanceSuccessFragmentModule.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+import dagger.Binds
+import dagger.Module
+import dagger.multibindings.IntoMap
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelKey
+
+@Module
+abstract class DccReissuanceSuccessFragmentModule {
```
[the remark this answers is not in the document]
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/success/DccReissuanceSuccessFragmentModule.kt:11]: Yes

File: Corona-Warn-App/src/main/res/navigation/covid_certificates_graph.xml [hunk 5]
```diff
@@ -296,4 +299,9 @@
...
+        android:name="de.rki.coronawarnapp.dccreissuance.ui.consent.DccReissuanceConsentFragment"
+        android:label="fragment_dcc_reissuance_consent"
+        tools:layout="@layout/fragment_dcc_reissuance_consent"></fragment>
```
[the remark this answers is not in the document]
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/res/navigation/covid_certificates_graph.xml]: Added an action to navigate to the SuccessFragment, so I need to leave it as is.

File: Corona-Warn-App/src/main/res/navigation/covid_certificates_graph.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/res/navigation/covid_certificates_graph.xml]: Renamed and added SuccessFragment.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/DccReissuanceModule.kt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package de.rki.coronawarnapp.dccreissuance
+
+import dagger.Module
+
+@Module
+class DccReissuanceModule
```
[the remark this answers is not in the document]
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/DccReissuanceModule.kt:6]: Done 😅

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/DccReissuanceUiModule.kt [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+import dagger.Module
+import dagger.android.ContributesAndroidInjector
+import de.rki.coronawarnapp.dccreissuance.ui.consent.DccReissuanceConsentFragment
+import de.rki.coronawarnapp.dccreissuance.ui.consent.DccReissuanceConsentFragmentModule
+import de.rki.coronawarnapp.dccreissuance.ui.success.DccReissuanceSuccessFragment
+import de.rki.coronawarnapp.dccreissuance.ui.success.DccReissuanceSuccessFragmentModule
+
+@Module
+abstract class DccReissuanceUiModule {
```
[the remark this answers is not in the document]
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/DccReissuanceUiModule.kt:11]: Added it to `MainActivityComponent`, is this correct?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/core/RecoveryCertificateRepository.kt [hunk 8]
```diff
@@ -277,6 +277,13 @@ class RecoveryCertificateRepository @Inject constructor(
         }
     }
 
+    suspend fun replaceCertificate(
+        certificateToReplace: RecoveryCertificateContainerId,
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/core/RecoveryCertificateRepository.kt:281]: Wouldn't it be easier to identify both with the QR code? The reissuance data structure has the qr code

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/core/RecoveryCertificateRepository.kt:279]: Both are fine , at the end we will `find ` the certificate by either fields , `RecoveryCertificateContainerId ` has the `hash` anyway

**Review by chiljamgossow — APPROVED** [review]

**Review by mtwalli — APPROVED** [review]

**Pull request merged**